ts/client/scripts/keeper/keeper.ts

```diff
@@ -0,0 +1,211 @@
+          );
+        }
+      }
+    }
+    await new Promise((r) => setTimeout(r, INTERVAL_CONSUME_EVENTS * 1000));
+  }
+}
+
+async function updateFunding(client: MangoClient, group: Group): Promise<void> {
+  console.log('Starting updateFunding loop');
+  // eslint-disable-next-line no-constant-condition
+  while (true) {
+    const perpMarketIndices = Array.from(
+      group.perpMarketsMapByMarketIndex.keys(),
+    );
+    for (const perpMarketIndex of perpMarketIndices) {
+      try {
+        const sig = await sendTransaction(
+          client.program.provider as AnchorProvider,
+          [
+            await client.perpUpdateFundingIx(
+              group,
+              group.getPerpMarketByMarketIndex(perpMarketIndex),
+            ),
+          ],
+          group.addressLookupTablesList,
+          { prioritizationFee: true },
+        );
+
+        console.log(
+          ` - Update funding success, perpMarketIndex - ${perpMarketIndex}, sig https://explorer.solana.com/tx/${sig}`,
+        );
+      } catch (e) {
+        console.log(
+          ` - Update funding error, perpMarketIndex - ${perpMarketIndex}, e - ${e}`,
+        );
+      }
+    }
+
+    await new Promise((r) => setTimeout(r, INTERVAL_UPDATE_FUNDING * 1000));
+  }
+}
+
+async function checkNewListingsAndAbort(
+  client: MangoClient,
+  group: Group,
+): Promise<void> {
+  console.log('Starting checkNewListingsAndAbort loop');
+  // eslint-disable-next-line no-constant-condition
+  while (true) {
+    const freshlyFetchedGroup = await client.getGroup(group.publicKey);
+    if (
+      freshlyFetchedGroup.banksMapByTokenIndex.size !=
+        group.banksMapByTokenIndex.size ||
+      freshlyFetchedGroup.perpMarketsMapByMarketIndex.size !=
+        group.perpMarketsMapByMarketIndex.size
+    ) {
+      process.exit();
+    }
+    await new Promise((r) =>
+      setTimeout(r, INTERVAL_CHECK_NEW_LISTINGS_AND_ABORT * 1000),
+    );
+  }
+}
+
+async function keeper(): Promise<void> {
+  // Load client
+  const options = AnchorProvider.defaultOptions();
+  const connection = new Connection(CLUSTER_URL!, options);
+  const user = Keypair.fromSecretKey(
+    Buffer.from(
+      JSON.parse(
+        process.env.KEYPAIR || fs.readFileSync(USER_KEYPAIR!, 'utf-8'),
+      ),
+    ),
+  );
+  const userWallet = new Wallet(user);
+  const userProvider = new AnchorProvider(connection, userWallet, options);
+  const client = await MangoClient.connect(
+    userProvider,
+    CLUSTER,
+    MANGO_V4_ID[CLUSTER],
+    {
+      idsSource: 'get-program-accounts',
+    },
+  );
+
+  const mangoAccount = await client.getMangoAccount(
+    new PublicKey(MANGO_ACCOUNT_PK),
+  );
+  const group = await client.getGroup(mangoAccount.group);
+  await group.reloadAll(client);
+
+  updateBanks(client, group);
+  consumeEvents(client, group);
+  updateFunding(client, group);
+  checkNewListingsAndAbort(client, group);
```

Comment:
does this work? you don't need to await the promises returned here?

Reply:
yes it works, we dont await the promises, this way we can basically trigger all functions

Reply:
already deployed at `heroku logs -a mango-v4-ts-keeper-1 -t`